feat(tooltip,tourtip,infotip): added popper support

agliga opened the pull request:
Fixes #1424

- [X] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
* Added support for popperjs. Changed all out examples to use popper
* I removed the use of makeup expander, since all it seems like it was doing was to toggle `aria-expanded`. We can revisit this if needed, but it doesn't seem necessary.
* I did not touch any storybooks since they are static. The old code should work as if if the users opt not to use popper (so there is no breaking changes)

## Checklist
- [X] I verify the build is in a non-broken state
- [X] I verify all changes are within scope of the linked issue
- [X] I regenerated all CSS files under dist folder
- [X] I tested the UI in all supported browsers
- [X] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [X] I tested the UI in dark mode and RTL mode
- [X] I added/updated/removed Storybook coverage as appropriate

ianmcburnie commented:
Just noticed this on the popper documentation:

> Safari has unfortunate quirks regarding updating the position of DOM elements. We have filed WebKit bugs for these, so hopefully they will be fixed in the future.

Might be worth looking at those issues to make sure we are comfortable with them.

ArtBlue commented on docs/_includes/infotip.html:
```diff
@@ -13,7 +14,7 @@
                 </button>
                 <div class="infotip__overlay">
-                    <span class="infotip__pointer infotip__pointer--top-left"></span>
+                    <span data-popper-arrow></span>
```
Wouldn't this be a breaking change?

agliga replied:
Because this is only a docs change. The old code is still available for teams to use.

ArtBlue replied:
Oh, right. Cool.

LuLaValva replied:
It won't be a breaking change for skin, but if we do this in eBayUI too we should tread lightly because there are a _lot_ of teams who override behavior of popovers to make them open in different directions.

agliga replied:
If they do that that's on them. They shouldn't be overriding base behavior 
But we do allow style overrides there. We probably should do something like
If they have a style override don't use popper to position.
If not then use popper.

ArtBlue replied:
I would prefer we not go down the road of checking for overrides and only using popper when none exist. I'd rather avoid various cases based on specific conditions if we can help it. If it comes down to it, I would actually prefer releasing this in a major version and warning people to remove overrides as part of the SWU.

agliga replied:
So I think there is a bit of misconception. First off this change would be only in ebayui. In skin there is no issue. In fact the old code works as is no problem (see storybook). 
This would only affect ebayui.

For ebayui I never said to not use popper if we have style overrides. We just don't use poppers automatic positioning. 
Popper allows you to override the position and setting offsets. What we would do in that case if they pass styles is to pass the position changes to popper and that would take care of it. 
So there wouldn't be anything breaking about it. It only only change internal implementation.

ArtBlue replied:
This is probably technically not a breaking change. The offsets are probably not an issue either; they serve as slight nudges. The positioning and overrides are the main thing I think.

Would we automatically send their styles into Popper without any checks? Or would we need to filter out the positioning ones and only send those to Popper? Would there be any sort of conditional overhead inside? If they override Popper's positioning, then the onus is on them to maintain it, right?

The main thing, I'd like to avoid is unnecessary positioning considerations being handled in ebay ui that later needs to be deprecated.

It's a bit hard conceptualizing this. Maybe if we had the ebay ui implementation, it might be easier to see if this poses any potential issue.

agliga commented:
> > Just noticed this on the popper [documentation](https://popper.js.org/docs/v2/browser-support/#safari):
> 
> > 
> 
> > > Safari has unfortunate quirks regarding updating the position of DOM elements. We have filed WebKit bugs for these, so hopefully they will be fixed in the future.
> 
> > 
> 
> > Might be worth looking at those issues to make sure we are comfortable with them. We have to support back to Safari 12.
> 
> 
> 
> Those issues look to be related to scrolling, one of them over-scrolling (rubber band) effect. I can see why they would occur. If the intent of the library is to attach one element to another positionally and also to avoid viewport boundaries and reposition itself within the viewport, I can see how scrolling would create a "conflict of interest" here.

Yah from what i see that on safari it would act just like what we had before. So it wont be any different from our previous implementation of tip overlays.

ArtBlue reviewed (dismissed):
LGTM.

ianmcburnie commented:
Is`[data-popper-placement]` added with JavaScript? Or is it declared in the base markup? I'm guessing the former as I don't see the attribute displayed in the example code. Which leads me to wonder whether we should be approaching this in a progressive enhancement fashion. Our existing modifiers and CSS work for most simple cases, so I don't think we need to always throw that away. It gives us a pretty good sensible default with the need for JS. Perhaps instead we should be progressively enhancing our existing modifiers (i.e. switching out `tooltip__pointer tooltip__pointer--left` for the popper equivalent). Let's chat offline about this.

ianmcburnie commented:
Can we please add a test story for popper overlay inside of container with overflow hidden.

ianmcburnie commented:
Not seeing the overlay or pointer appear in this example in Safari:

<img width="303" alt="Screenshot 2023-10-03 at 12 56 35 PM" src="https://github.com/eBay/skin/assets/38065/1a2e8463-f2ce-4c6c-8a37-cfc0d8e976c7">

Here it is in Chrome:

<img width="253" alt="Screenshot 2023-10-03 at 12 57 14 PM" src="https://github.com/eBay/skin/assets/38065/55cfd5f2-fbf4-4b70-ada7-27e997209d06">

agliga commented:
Will wait for next release to get all the cleanup.

ArtBlue reviewed (approved):
Looks good!

ianmcburnie commented:
On tourtips, the pointers look broken (in safari & chrome):
<img width="709" alt="Screenshot 2023-12-18 at 11 26 42 AM" src="https://github.com/eBay/skin/assets/38065/4314819f-5032-4c7b-bb5d-a18a0f67bdfd">
<img width="701" alt="Screenshot 2023-12-18 at 11 26 36 AM" src="https://github.com/eBay/skin/assets/38065/fd5419a6-8be4-4840-89da-dea29959acf9">

ianmcburnie commented:
Isn't this Infotip pointer supposed to flip from bottom to top?

<img width="615" alt="Screenshot 2023-12-18 at 11 29 25 AM" src="https://github.com/eBay/skin/assets/38065/c9d40593-b637-4071-915a-744579dbb41a">


<img width="554" alt="Screenshot 2023-12-18 at 11 29 36 AM" src="https://github.com/eBay/skin/assets/38065/66b73e98-e57a-4c6d-8964-a7404496c580">

agliga commented:
> Isn't this Infotip pointer supposed to flip from bottom to top when the info top repositions itself?
> 
> <img alt="Screenshot 2023-12-18 at 11 29 25 AM" width="615" src="https://private-user-images.githubusercontent.com/38065/291388713-c9d40593-b637-4071-915a-744579dbb41a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5NDMzMzksIm5iZiI6MTcwMjk0MzAzOSwicGF0aCI6Ii8zODA2NS8yOTEzODg3MTMtYzlkNDA1OTMtYjYzNy00MDcxLTkxNWEtNzQ0NTc5ZGJiNDFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE4VDIzNDM1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThmMjdiNWY1YzU0ZTE4NDhiYWNhNDI1MzQyMGE4N2I2OTIwYjZiMjA0ZTMzNmZhYWFiODQ2MmQ4MzQ3MGUwMzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lfLIkiCOQSCUbgWLuL8_JFKbiubZ7KJ-qfMaVfc4ZSY"> <img alt="Screenshot 2023-12-18 at 11 29 36 AM" width="554" src="https://private-user-images.githubusercontent.com/38065/291388698-66b73e98-e57a-4c6d-8964-a7404496c580.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5NDMzMzksIm5iZiI6MTcwMjk0MzAzOSwicGF0aCI6Ii8zODA2NS8yOTEzODg2OTgtNjZiNzNlOTgtZTU3YS00YzZkLTg5NjQtYTc0MDQ0OTZjNTgwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE4VDIzNDM1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMzNTUyZjVmMzYxNTRjMjE0ODEyYTdiNGFiYjNlZjZmZmZhZmFhNWYyMmZlMDk0NWI0MDdkYTg5ZDBhYzBiMjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.J0LZkjoC9fR1IXJ4ECNJjPkK6XuMY45naUMZKyQN4fk">

Fixed, this was due to doing `!==` instead of `!=` as the docs recommended.

agliga commented:
> On tourtips, the pointers look broken (in safari & chrome): <img alt="Screenshot 2023-12-18 at 11 26 42 AM" width="709" src="https://private-user-images.githubusercontent.com/38065/291388316-4314819f-5032-4c7b-bb5d-a18a0f67bdfd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5NDMzMzksIm5iZiI6MTcwMjk0MzAzOSwicGF0aCI6Ii8zODA2NS8yOTEzODgzMTYtNDMxNDgxOWYtNTAzMi00YzdiLWJiNWQtYTE4YTBmNjdiZGZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE4VDIzNDM1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI1ZDE4NTRmNTc0ZDI2ODVmNDBmMDc1ZGY4MWRiOTJiZTNlMDgyOWM1Y2I0ZGQxYTk3YmFiOTg3MjlmZWNkZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.K12fLmfhaOB9YimD1QvivFUaS179vd9qjmORBt1gLaE"> <img alt="Screenshot 2023-12-18 at 11 26 36 AM" width="701" src="https://private-user-images.githubusercontent.com/38065/291388334-fd5419a6-8be4-4840-89da-dea29959acf9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5NDMzMzksIm5iZiI6MTcwMjk0MzAzOSwicGF0aCI6Ii8zODA2NS8yOTEzODgzMzQtZmQ1NDE5YTYtOGJlNC00ODQwLTg5ZGEtZGVhMjk5NTlhY2Y5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE4VDIzNDM1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc0Y2EwZjNlNjY2MDBlYTE0OGFhNzJiZDQ4MGI1Y2Y3M2U2NzVjNmRkMDRjNDI4MGU3ZTVlOWM0NWU5MDE3YTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7tH_lcaaGjeL_GpMMcEVdDvobMsQDZbdUy101hv6vXc">

Added a 20px padding on arrow (The padding on the arrow is based on the border radius values so it doesn't look like this).

LuLaValva reviewed (approved):
Looks awesome! I'm excited to see this on other floating elements like `listbox` and `date-textbox`.

It would have probably been nice to do prettier for `main.js` in a separate PR but I'll let it slide

ianmcburnie reviewed (changes requested):
I would like to still keep the examples of infotip, tooltip, tourtip that we had _without_ popper, and then added a dynamic positioning section for each which shows what accommodations we have to make in the markup/css (if any) for a 3rd party module like popper.

ianmcburnie commented on docs/_includes/tooltip.html:
```diff
@@ -4,6 +4,7 @@
     <p>A tooltip gives additional information about an interactive element - typically a button. The tooltip activates on mouse hover or keyboard focus of the button.</p>
     <p>The tooltip must be programmatically associated with the button by using the <span class="highlight">aria-describedby</span> property and <span class="highlight">tooltip</span> role.</p>
     <p>Toggle the <span class="highlight">aria-expanded</span> state of the button to expand or collapse its associated tooltip.</p>
+    <p>For this first example this is using CSS positioning and works without JS</p>
```
```suggestion
```

ianmcburnie commented on docs/_includes/tooltip.html:
```diff
@@ -46,22 +47,23 @@
 
-    <h3 id="tooltip-position">Positioning the Tooltip</h3>
-    <p>The tooltip pointer modifier supports 8 different positions. In clockwise order, moving around the interactive element that is being pointed at, they are: bottom, bottom-left, left, top-left, top, top-right, right, bottom-right.</p>
+    <p>For these following examples, we are using floating-ui to position the tooltip and the arrow.</p>
```
```suggestion
    <p>By default, the position of the tooltip and its pointer will remain static. For these following examples, we are using [floating-ui](https://floating-ui.com) to help position the tooltip and the pointer _dynamically_.</p>
```

ianmcburnie commented on docs/_includes/infotip.html:
```diff
@@ -3,17 +3,74 @@
 
     <p>An infotip is a button which can be clicked to display information about another element or area on the page.</p>
     <p>Toggle the <span class="highlight">aria-expanded</span> state of the button to expand or collapse the infotip.</p>
+    <p>For this first example this is using CSS positioning and works without JS</p>
```
```suggestion
```

ianmcburnie commented on docs/_includes/infotip.html:
```diff
@@ -3,17 +3,74 @@
+</span>
+    {% endhighlight %}
+
+    <p>For these following examples, we are using floating-ui to position the infotip and the arrow.</p>
```
```suggestion
    <p>By default, the position of the overlay and its pointer will remain static. For these following examples, we are using [floating-ui](https://floating-ui.com) to help position the tooltip and the pointer _dynamically_.</p>
```

ianmcburnie commented on docs/_includes/tourtip.html:
```diff
@@ -3,13 +3,14 @@
 
     <p>A tourtip points out a new feature or section of the page. A tourtip is open by default and must be explcitly closed using its close button. Once closed, it cannot be reopened.</p>
     <p>Toggle the <span class="highlight">tourtip--expanded</span> modifier to expand or collapse the tourtip.</p>
+    <p>For this first example this is using CSS positioning and works without JS</p>
```
```suggestion
```

ianmcburnie commented on docs/_includes/tourtip.html:
```diff
@@ -30,8 +31,56 @@ <h2 class="tourtip__heading" id="tourtip-label">Tourtip</h2>
+</div>
+    {% endhighlight %}
+
+    <p>For this example, we are using floating-ui to position the tourtip and the arrow</p>
```
```suggestion
    <p>By default, the position of the overlay and its pointer will remain static. For these following examples, we are using [floating-ui](https://floating-ui.com) to help position the tooltip and the pointer _dynamically_.</p>
```

ianmcburnie commented:
Noticing a couple of discrepancies between positioning on prod storybook vs branch in tooltip stories:

<img width="373" alt="Screenshot 2023-12-20 at 3 14 00 PM" src="https://github.com/eBay/skin/assets/38065/02e31c0b-8043-48fb-ae4c-db019e20712c">
<img width="390" alt="Screenshot 2023-12-20 at 3 13 49 PM" src="https://github.com/eBay/skin/assets/38065/882d1ce3-ebf7-46c8-a5ce-02158a242eba">

<img width="243" alt="Screenshot 2023-12-20 at 3 15 30 PM" src="https://github.com/eBay/skin/assets/38065/3f12bd1c-dac3-4783-b149-1acd07dbddec">
<img width="265" alt="Screenshot 2023-12-20 at 3 15 27 PM" src="https://github.com/eBay/skin/assets/38065/bccb34fa-8c9c-42da-824f-aee695aee12c">

ianmcburnie commented on docs/_includes/infotip.html:
```diff
@@ -7,13 +7,69 @@
+</span>
+    {% endhighlight %}
+
+    <p>By default, the position of the overlay and its pointer will remain static. For these following examples, we are using [floating-ui](https://floating-ui.com) to help position the tooltip and the pointer _dynamically_.</p>
```
Oops! That was my bad.

agliga replied:
Fixed this.

agliga commented:
> Noticing a couple of discrepancies between positioning on prod storybook vs branch in tooltip stories:
> 
> tooltip -> expanded
> 
> <img alt="Screenshot 2023-12-20 at 3 14 00 PM" width="373" src="https://private-user-images.githubusercontent.com/38065/292045387-02e31c0b-8043-48fb-ae4c-db019e20712c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxNzg2OTQsIm5iZiI6MTcwMzE3ODM5NCwicGF0aCI6Ii8zODA2NS8yOTIwNDUzODctMDJlMzFjMGItODA0My00OGZiLWFlNGMtZGIwMTllMjA3MTJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIxVDE3MDYzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVhNzE0ZTJhMjhjZWY2YzllNjNjYzhlMDI0YzUxZTJkMDEzNzM1MGMzMWYwMTU2MDJkNmIxNDkzNWIxNjNhM2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.YwgonVlc6irntDHO2nsnffKF1UzoUE5VKdI2ivdIdPI"> <img alt="Screenshot 2023-12-20 at 3 13 49 PM" width="390" src="https://private-user-images.githubusercontent.com/38065/292045390-882d1ce3-ebf7-46c8-a5ce-02158a242eba.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxNzg2OTQsIm5iZiI6MTcwMzE3ODM5NCwicGF0aCI6Ii8zODA2NS8yOTIwNDUzOTAtODgyZDFjZTMtZWJmNy00NmM4LWE1Y2UtMDIxNThhMjQyZWJhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIxVDE3MDYzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc4MGM2ZmMzZjc2MDQ3MjFhNWZiN2RlNmRhYmVhMmMxOGI3MTcyMDE2Yjg4MzJmMTY4N2E3ZTUyOTg5YzIwNTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jZQZICrhQJ5-LpAytj9E_wC7MMlNYSM0NRwhdr3lZNw">
> Tooltip -> pointer -> left (and all others)
> 
> <img alt="Screenshot 2023-12-20 at 3 15 30 PM" width="243" src="https://private-user-images.githubusercontent.com/38065/292045484-3f12bd1c-dac3-4783-b149-1acd07dbddec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxNzg2OTQsIm5iZiI6MTcwMzE3ODM5NCwicGF0aCI6Ii8zODA2NS8yOTIwNDU0ODQtM2YxMmJkMWMtZGFjMy00NzgzLWIxNDktMWFjZDA3ZGJkZGVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIxVDE3MDYzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEyZmRkMTAyZGZiMDJiM2Y4NmI4ZDY3NTIzMzVkMTRmMjBmNjc0Y2M4NTZkNDgwMDlkYzM3NTkwNzI5M2EzZjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._GB3cQpT74ShfVrspG4EqtPlSQ-wmrvxms4SZUT2NPI"> <img alt="Screenshot 2023-12-20 at 3 15 27 PM" width="265" src="https://private-user-images.githubusercontent.com/38065/292045487-bccb34fa-8c9c-42da-824f-aee695aee12c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxNzg2OTQsIm5iZiI6MTcwMzE3ODM5NCwicGF0aCI6Ii8zODA2NS8yOTIwNDU0ODctYmNjYjM0ZmEtOGM5Yy00MmRhLTgyNGYtYWVlNjk1YWVlMTJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIxVDE3MDYzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUyZDc5YjUxZmQxZmYwN2ExYzhmYTAyYzgwMTU5NTY5MGM3NzAwN2Q4Zjc3ZjMzN2MyMTdlZDQ2OTQ0OGM5OGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.wI0ubo9NzuvckDEw2XXsUXKHGtFUhBOaOwlckaazUis">

Fixed this. I had removed the margin on bubble which was breaking in other places. I changed the positioning of the tip to get it to work.

ArtBlue reviewed (commented):
Looks fine to me, but we should get a confirmation approval from Ian since he had the bulk of feedback.

ianmcburnie reviewed (approved):
Great work @agliga
